ras_party/views/pending_survey_view.py

```diff
@@ -200,8 +202,87 @@ def confirm_pending_shares(batch_no):
 @pending_survey_view.route("/pending-surveys/<batch_no>", methods=["GET"])
 def get_pending_surveys_with_batch_no(batch_no):
     """
-    Confirms pending share survey
+    Get pending surveys by batch number
     :param batch_no
     """
     response = get_pending_survey_by_batch_number(batch_no)
     return make_response(jsonify(response), 200)
+
+
+@pending_survey_view.route("/pending-surveys/originator/<originator_respondent_party_id>", methods=["GET"])
+def get_pending_surveys_with_originator_respondent_party_id(originator_respondent_party_id):
+    """
+    Get pending surveys record against originator party id
+    :param originator_respondent_party_id: Respondent party id
+    :type originator_respondent_party_id: uuid
+    """
+    logger.info(
+        "Retrieving share survey records against originator",
+        originator_respondent_party_id=originator_respondent_party_id,
+    )
+    response = get_pending_survey_by_originator_respondent_id(originator_respondent_party_id)
+    return make_response(jsonify(response), 200)
+
+
+@pending_survey_view.route("/pending-surveys/<batch_no>", methods=["DELETE"])
+def delete_pending_surveys_by_batch_number(batch_no):
+    """
+    Delete pending surveys record against a batch_number
+    :param batch_no: batch number
+    :type batch_no: uuid
+    """
+    logger.info("Attempting to delete pending surveys record against batch number", batch_no=batch_no)
+    response = pending_survey_deletion(batch_no)
+    return response
+
+
+@pending_survey_view.route("/pending-surveys/resend-email", methods=["POST"])
+def resend_pending_surveys_email():
+    """
+    Request to resend pending share email address against a single batch_number
+    accepted payload:
+    {
+        'batch_number': #batch_number
+    }
+    """
+    logger.info("Attempting to resend email for pending survey")
+    payload = request.get_json() or {}
+    if "batch_number" not in payload:
+        logger.error("Invalid request")
+        raise BadRequest("Invalid request - batch_number missing")
+    batch_number = payload["batch_number"]
+    logger.info("retrieving pending surveys against batch number", batch_number=batch_number)
+    response = get_pending_survey_by_batch_number(batch_number)
```

Comment:
`response` isn't a great name for this I feel. maybe `pending_surveys` ? or something like that